From 4d8a3e34fb190cd019e993015f00914aaf270d67 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Tue, 6 Apr 2021 18:38:50 +0000
Subject: [PATCH] GCC: fix ternary operator in gtk_util

Ternary operator requires that operands after ? have
same type, but is std::nullptr_t and ScopedGObject<GtkWidget>.
Use ScopedGObject<GtkWidget>() for std::nullptr_t instead.
---
 ui/gtk/gtk_util.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui/gtk/gtk_util.cc b/ui/gtk/gtk_util.cc
index f1cc779..9c08fa4 100644
--- a/ui/gtk/gtk_util.cc
+++ b/ui/gtk/gtk_util.cc
@@ -285,7 +285,7 @@ GtkCssContext::operator GtkStyleContext*() {
 GtkCssContext GtkCssContext::GetParent() {
   if (GtkCheckVersion(4)) {
     return GtkCssContext(WrapGObject(gtk_widget_get_parent(widget_)),
-                         root_ == widget_ ? nullptr : root_);
+                         root_ == widget_ ? ScopedGObject<GtkWidget>() : root_);
   }
   return GtkCssContext(WrapGObject(gtk_style_context_get_parent(context_)));
 }
-- 
2.26.3

